ledfx/dedupequeue.py

```
@@ -0,0 +1,29 @@
+            if any(self.is_similar(item, existing_item) for existing_item in self._queue):
+                # TODO: remove this logging
+                _LOGGER.warning(f"Queue: {hex(id(self))} discarding, qsize {self.qsize()}")
+                return
+        super().put_nowait(item)
+
+    def is_similar(self, new, queued):
+        # We know we are already one of the correct types, but is it the same as queued
+        # then check if it is for the same device
+        if new.get("event_type") == queued.get("event_type") and new.get("vis_id") == queued.get("vis_id"):
+            return True
+        return False
```

Ruff

Return the condition directly

Inline condition

(SIM103)